### DISCUSSION TITLE
What is the difference b/w `shortcuts` and `@layer components`?

### DISCUSSION THREAD

**Discussion by ManasMadrecha** [windicss/windicss#598] [Q&A] [answered]
Using `shortcuts` features, we can create reusable components so that we need not type all the classes related to it again and again. 

But, using `@layer components`, we can do the same, can't we?

So, what is the difference between the two? And which should be preferred?

**Comment by antfu** [comment, marked as answer]
Just use the one that makes sense to you. Shortcuts are defined in JS meaning you can have custom logic to generate them as you want. Shortcuts also supports applying variants.

**Reply by ManasMadrecha** [reply]
Migrating an existing from Tailwind, so I already have SASS files written using `@layer components`. But, for a new project, is it preferable to use `shortcuts` in `windi.config.js` file? What sort of custom logic can be used there? Can you kindly provide an example?

**Reply by antfu** [reply]
For example, you can write 

```ts
shortcuts: {
  ...Object.fromEntries(Object.key(colors).map(c => [`button-${c}`, `.... bg-${c}`]))
}
```

to support all colors of your custom button. And meanwhile, you can even extract that to a separate package and share it between projects.

**Reply by ManasMadrecha** [reply]
Wow, this seems great! 😮 Thanks for the guidance.
